static/app/views/performance/database/useDenylistedProjects.tsx

```diff
@@ -0,0 +1,36 @@
+import useProjects from 'sentry/utils/useProjects';
+
+interface Options {
+  enabled?: boolean;
+  projectId?: string[];
+}
+
+/**
+ * Returns a list of projects that are not eligible for span metrics
+ * because they were denylisted.
+ *
+ * @param options Additional options
+ * @param options.projectId List of project IDs to check against. If omitted, checks all organization projects
+ * @returns List of projects
+ */
+export function useDenylistedProjects(options?: Options) {
+  const {projects, fetching} = useProjects();
+
+  const shouldCheckAllProjects =
+    options?.projectId?.length === 0 || options?.projectId?.includes('-1');
```

Comment:
l: Isn't it enough to only do `options?.projectId?.includes` check?

Another nit, I find these kind of conditionals hard to read because we have a mixture of `undefined` + `boolean` and coercion is hard to follow. Can we do early returns if `options` or `options.projectId` is `undefined` and then deal with fully validated data later on?

Reply:
I don't think that's enough, the app behaves the same whether project ID is explicitly `-1` or missing completely in the URL. There are similar checks in other parts of Performance.

There's also no early return to be done here, since none of the conditions are invalid, they just have different code paths

If you have thoughts on making this easier to read, I'm all ears! I personally find that if I squint and pretend the `?.` aren't there it reads okay. I can assign the conditions to variables, and their name would clarify, maybe?

Reply:
Thinking about this more, rather than early returns, I think default values improves the readability for me.

Something like so:

```ts
export function useDenylistedProjects(options: Options = {}) {
  const {projects, fetching} = useProjects();

  const { projectId = [] } = options;

  const shouldCheckAllProjects = projectId.length === 0 || projectId.includes('-1');
  // ...
```

my opinion is that propagating down `undefined` means I have the mental overhead when reading to understand what happens if optional chaining causes a shortcircuit and coerces the value to be `undefined`.

Reply:
👍🏻 works for me